python-threatexchange/threatexchange/signal_type/signal_base.py

```diff
@@ -95,7 +95,21 @@ def get_examples() -> t.List[str]:
         return []
 
 
-class TextHasher:
+class FileHasher:
+    """
+    This class can hash files.
+
+    If also inheriting from StrHasher, put this second in the inheritance
+    to prefer file hashing to reading the file in as a Str.
+    """
+
+    @classmethod
+    def hash_from_file(cls, file: pathlib.Path) -> str:
+        """Get a string representation of the hash from a file"""
+        raise NotImplementedError
+
+
+class TextHasher(FileHasher):
```

Comment:
Hmm, thought this was already the case. 

Ah derp, it just implements the function without actually extending it. Yay python.